python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1274,16 +1274,22 @@ def get_canceling_asset_backfill_iteration_data(
         )
 
     failed_subset = AssetGraphSubset.from_asset_partition_set(
-        set(_get_failed_asset_partitions(instance_queryer, backfill_id, asset_graph)),
+        set(
+            _get_failed_asset_partitions(
+                instance_queryer,
+                backfill_id,
+                asset_graph,
+                materialized_subset=updated_materialized_subset,
+            )
+        ),
         asset_graph,
     )
     updated_backfill_data = AssetBackfillData(
         target_subset=asset_backfill_data.target_subset,
         latest_storage_id=asset_backfill_data.latest_storage_id,
         requested_runs_for_target_roots=asset_backfill_data.requested_runs_for_target_roots,
         materialized_subset=updated_materialized_subset,
-        failed_and_downstream_subset=asset_backfill_data.failed_and_downstream_subset
-        | failed_subset,
+        failed_and_downstream_subset=failed_subset,
```

Comment:
Want to confirm that this is a safe change to make. From my reading of `_get_failed_asset_partitions` it returns the full list of failed partitions, not a list of partitions that failed since the last tick, so the existing ORing of the two subsets is a no-op since `asset_backfill_data.failed_and_downstream_subset` would be a subset of `failed_subset`

with the change to have `_get_failed_asset_partitions` account for retries, ORing with `asset_backfill_data.failed_and_downstream_subset` would result in inaccurate data because a failed partition in `asset_backfill_data.failed_and_downstream_subset` could have been successfully retried and no longer in `failed_subset` but would still be included because of the OR operation

Reply:
ah ok i forgot that we do the second function to get the downstream of the failed subset, so this isn't a simple replacement since `failed_subset` doesn't include downstream assets. i will update